### PR TITLE
[enhance](mtmv)Mtmv datetrunc support week and quarter

### PR DESCRIPTION
**Pull request by zddr**
if base table have 3 partitions:
```
 PARTITION p_20200101 VALUES [("2020-01-01"),("2020-01-02")),
  PARTITION p_20200102 VALUES [("2020-01-02"),("2020-01-03")),
  PARTITION p_20200108 VALUES [("2020-01-08"),("2020-01-09"))
```
mtmv partition date_trunc(week)

mtmv will have 2 partitions
```
[("2019-12-30"),("2020-01-06"))
[("2020-01-06"),("2020-01-13"))
```

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run cloud_p1

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionExprDateTrunc.java [hunk 1]
```diff
@@ -198,9 +198,15 @@ private DateTimeV2Literal dateIncrement(DateTimeV2Literal value) throws Analysis
             case "year":
                 result = value.plusYears(1L);
                 break;
+            case "quarter":
+                result = value.plusMonths(3L);
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionExprDateTrunc.java:202]: why add 3 month, can add some comment to explain this

Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionExprDateTrunc.java:202]: 1 quarter = 3month

Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionExprDateTrunc.java:202]: value not have `plusQuarter` method,so use plusMonths(3L);

**Review by seawinde — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**